### ISSUE TITLE
DynamoAttributeValueMap implementation in csharp

### ISSUE THREAD

**Issue by whoiskevinrich** [aws/aws-cdk#7884]
## :question: General Issue
DynamoAttributeValueMap implementation in csharp seems to be missing

### The Question
My colleagues and I are attempting to write a step-function with a DynamoPutItem Task in csharp. We're having an issue building the `Item` property. I could find no documentation for constructing a `DynamoAttributeValueMap` object for the property and it seems to have no properties or methods.

Is this an issue with the csharp CDK or are we using it incorrectly?

### Environment

  - **CDK CLI Version:** 1.38.0
  - **Module Version:** 1.38.0
  - **OS:** Windows 10 | Ubuntu
  - **Language:** CSharp

**Comment by shivlaks** [comment]
@whoiskevinrich - I haven't worked with the dynamodb service integration much yet. It could very well be a bug. To help narrow this down and triage, can you share some code and the behaviour you're observing vs. the behaviour you would expect.

**Comment by whoiskevinrich** [comment]
Example Task:

```csharp
new Task(scope, "MyWriteTaskId", new TaskProps
{
  Task = CallDynamoDB.PutItem(new DynamoPutItemProps
  {
    TableName = "MyTable",
    Item = new DynamoAttributeValueMap() 
  }
}
```

This compiles, but there are no available methods or properties on the `DynamoAttributeValueMap` object to configure the mapping with. 

In reading the Dynamo API docs for [GetItem](https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_GetItem.html) and [PutItem](https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_PutItem.html#API_PutItem_RequestSyntax) it would seem that the Item property could take a set of DynamoAttributes.

I would expect to see something similar to this:

```csharp
new Task(scope, "MyWriteTaskId", new TaskProps
{
  Task = CallDynamoDB.PutItem(new DynamoPutItemProps
  {
    TableName = "MyTable",
    Item = new DynamoAttributeValueMap[]
    {
      new DynamoAttribute
      {
        Name = "MyAttributeName",
        Value = new DynamoAttributeValue().WithS("MyNewValue")
      }
    }
  }
}
```

**Comment by shivlaks** [comment]
@whoiskevinrich - you're absolutely right, it's not usable in Python, C#, or Java the way it is currently modeled.
PR with fix is in review #8466

**Comment by shivlaks** [comment]
resolving this issue as the `DynamoDB` integration with Step Functions was re-written in #8466
feel free to re-open if there are any outstanding issues though!
